core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -63,6 +63,20 @@ public static boolean ancestorOf(Table table, long snapshotId, long ancestorSnap
     return ancestorIds(table.currentSnapshot(), table::snapshot);
   }
 
+  /**
+   * Find the oldest Snapshot of a table.
+   * @param table the table to find the oldest snapshot on.
+   * @return null if the table is empty, else the oldest Snapshot.
+   */
+  public static Snapshot oldestSnapshot(Table table) {
+    Snapshot current = table.currentSnapshot();
+    while (current.parentId() != null) {
```

Comment:
NPE? If currentSnapshot() is null then we would npe here calling parentID

Reply:
nice catch. thanks. fixed it.